proparse/src/main/java/org/prorefactor/proparse/antlr4/Lexer.java

```diff
@@ -219,56 +219,61 @@ ProToken nextToken() throws IOException {
           if (currChar == '=') {
             append();
             getChar();
-            return makeToken(ProParserTokenTypes.GTOREQUAL);
+            return makeToken(PreprocessorParser.GTOREQUAL);
           } else {
-            return makeToken(ProParserTokenTypes.RIGHTANGLE);
+            return makeToken(PreprocessorParser.RIGHTANGLE);
           }
 
         case '<':
           getChar();
           if (currChar == '>') {
             append();
             getChar();
-            return makeToken(ProParserTokenTypes.GTORLT);
+            return makeToken(PreprocessorParser.GTORLT);
           } else if (currChar == '=') {
             append();
             getChar();
-            return makeToken(ProParserTokenTypes.LTOREQUAL);
+            return makeToken(PreprocessorParser.LTOREQUAL);
           } else {
-            return makeToken(ProParserTokenTypes.LEFTANGLE);
+            return makeToken(PreprocessorParser.LEFTANGLE);
           }
 
         case '+':
           getChar();
-          return plusMinusStart(ProParserTokenTypes.PLUS);
+          return plusMinusStart(PreprocessorParser.PLUS);
         case '-':
           getChar();
-          return plusMinusStart(ProParserTokenTypes.MINUS);
+          return plusMinusStart(PreprocessorParser.MINUS);
 
         case '#':
         case '|':
         case '%':
           getChar();
-          return id(ProParserTokenTypes.FILENAME);
+          return id(PreprocessorParser.FILENAME);
 
         default:
           if (currInt == EOF_CHAR) {
             getChar(); // preprocessor will catch any infinite loop on this.
-            return makeToken(ProParserTokenTypes.EOF, "");
+            return makeToken(PreprocessorParser.EOF, "");
           } else {
             getChar();
-            return id(ProParserTokenTypes.ID);
+            return id(PreprocessorParser.ID);
           }
 
       }
     }
+    } catch(IOException uncaught) {
+      throw new RuntimeException(uncaught);
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Define and throw a dedicated exception instead of using a generic one. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://sonar.riverside-software.fr/coding_rules#rule_key=squid%3AS00112)